test/e2e/dra/dra.go

```diff
@@ -1911,6 +1921,175 @@ var _ = framework.SIGDescribe("node")(framework.WithLabel("DRA"), framework.With
 		})
 	})
 
+	framework.Context(f.WithFeatureGate(features.DRAExtendedResource), "single node", func() {
+		nodes := NewNodes(f, 1, 1)
+		driver := NewDriver(f, nodes, networkResources(10, false))
+		b := newBuilder(f, driver)
+
+		ginkgo.It("must run a pod with extended resource", ginkgo.Label("dra extended resource single-node"), func(ctx context.Context) {
+			pod := b.pod()
+			res := v1.ResourceList{}
+			res[v1.ResourceName(extendedResourceName)] = resource.MustParse("1")
+			pod.Spec.Containers[0].Resources.Requests = res
+			pod.Spec.Containers[0].Resources.Limits = res
+
+			b.create(ctx, pod)
+			err := e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod)
+			framework.ExpectNoError(err, "start pod")
+			containerEnv := []string{
+				"container_0_request_0", "true",
+			}
+			testContainerEnv(ctx, f, pod, pod.Spec.Containers[0].Name, false, containerEnv...)
+		})
+
+		ginkgo.It("must run a pod with extended resource", ginkgo.Label("dra extended resource single-node three-extended-resources"), func(ctx context.Context) {
+			pod := b.pod()
+			res := v1.ResourceList{}
+			res[v1.ResourceName(extendedResourceName)] = resource.MustParse("1")
+			res[v1.ResourceName(extendedResourceName2)] = resource.MustParse("1")
+			res[v1.ResourceName(extendedResourceName3)] = resource.MustParse("1")
+			pod.Spec.Containers[0].Resources.Requests = res
+			pod.Spec.Containers[0].Resources.Limits = res
+
+			b.create(ctx, pod)
+			err := e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod)
+			framework.ExpectNoError(err, "start pod")
+			containerEnv := []string{
+				"container_0_request_0", "true",
+				"container_0_request_1", "true",
+				"container_0_request_2", "true",
+			}
+			testContainerEnv(ctx, f, pod, pod.Spec.Containers[0].Name, false, containerEnv...)
+		})
+		ginkgo.It("must run a pod with extended resource", ginkgo.Label("dra extended resource single-node three-containers"), func(ctx context.Context) {
+			pod := b.pod()
+			pod.Spec.Containers = append(pod.Spec.Containers, *pod.Spec.Containers[0].DeepCopy())
+			pod.Spec.Containers = append(pod.Spec.Containers, *pod.Spec.Containers[0].DeepCopy())
+			pod.Spec.Containers[0].Name = "container0"
+			pod.Spec.Containers[1].Name = "container1"
+			pod.Spec.Containers[1].Name = "container2"
```

Comment:
```suggestion
			pod.Spec.Containers[2].Name = "container2"
```

But really, please use a for loop instead to avoid such cut-and-paste errors. Same applies to several other places above and below.